vm/src/obj/objstr.rs

```diff
@@ -187,7 +247,7 @@ impl PyStringRef {
 
     fn split(
         self,
-        pattern: OptionalArg<Self>,
+        pattern: OptionalArg<PyStringRef>,
```

Comment:
I think you can go back to using `Self` in places like here too, right?

Reply:
Personally I prefer `PyStringRef` here. `Self` makes sense for an argument when you want to emphasize that you are taking another object of the same type, such as for `eq` and such, but here it's more of a coincidence that `pattern` happens to also be a string.

Reply:
Personally I prefer PyStringRef here. Self makes sense for an argument when you want to emphasize that you are taking another object of the same type, such as for eq and such, but here it's more of a coincidence that pattern happens to also be a string.

Reply:
:+1: